File: portfolio/src/main/java/dev/exit/asset/portfolio/PortfolioController.java
package dev.exit.asset.portfolio;

import java.time.YearMonth;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import dev.exit.asset.common.CurrentUserProvider;
import dev.exit.asset.portfolio.PortfolioSummaryService;
import dev.exit.asset.portfolio.PortfolioSummaryService.PortfolioSummary;

@RestController
@RequestMapping("/api/v1/portfolio")
public class PortfolioController {

	private final CurrentUserProvider currentUser;

	private final PortfolioSummaryService summaries;

	public PortfolioController(CurrentUserProvider currentUser, PortfolioSummaryService summaries) {
		this.currentUser = currentUser;
		this.summaries = summaries;
	}

	@GetMapping("/summary")
	public PortfolioSummary summary(@RequestParam(required = false) YearMonth month) {
		return summaries.summarize(currentUser.ownerId(), month == null ? YearMonth.now() : month);
	}
}

File: portfolio/src/main/kotlin/dev/exit/asset/portfolio/PortfolioController.kt
package dev.exit.asset.portfolio

import dev.exit.asset.common.CurrentUserProvider
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.time.YearMonth

@RestController
@RequestMapping("/api/v1/portfolio")
class PortfolioController(
    private val currentUser: CurrentUserProvider,
    private val summaries: PortfolioSummaryService,
) {

    @GetMapping("/summary")
    fun summary(@RequestParam(required = false) month: YearMonth?): PortfolioSummaryService.PortfolioSummary =
        summaries.summarize(currentUser.ownerId(), month ?: YearMonth.now())
}
